### PR TITLE
Update discovery examples to include device creation and script

### LINKED ISSUE #27989: The "device" section is required for discovery but it is not obvious from the documentation
### Feedback

The "device" section is needed for config-messages, otherwise home-assistant does not discover the device. I think this information is not very obvious in the documentation.

### URL

https://www.home-assistant.io/integrations/mqtt/

### Version

2023.6.3

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by raythefourth**
## Proposed change
Currently, [this example](https://www.home-assistant.io/integrations/mqtt/#sensors) does not produce the intended results.

This change will update the discovery examples section to use correct Home Assistant terminology for sensors, entities, and devices. Examples have been extended to include a useful script for automating device creation.

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #27989

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by jbouwh — CHANGES_REQUESTED** [review]
I think we should try to avoid changing titles. I'll hope to do a further review this weekend.

File: source/_integrations/mqtt.markdown [hunk 1]
```diff
@@ -559,44 +559,233 @@ The following software has built-in support for MQTT discovery:
 
 ### Discovery examples
 
-#### Motion detection (binary sensor)
+#### Creating a single sensor entity (motion detection binary sensor)
 
-A motion detection device which can be represented by a [binary sensor](/integrations/binary_sensor.mqtt/) for your garden would send its configuration as JSON payload to the Configuration topic. After the first message to `config`, then the MQTT messages sent to the state topic will update the state in Home Assistant.
+In the following example, a single entity will be created representing the state of a motion detection sensor. The motion detection sensor has only 2 states, therefore it can be represented by an [MQTT Binary Sensor](/integrations/binary_sensor.mqtt/).
```
Comment (jbouwh) [inline source/_integrations/mqtt.markdown:564]: I am not happy with changes like `single entity`. Further it is not needed to say a binary sensor has only 2 states

File: source/_integrations/mqtt.markdown [hunk 2]
```diff
@@ -559,44 +559,233 @@ The following software has built-in support for MQTT discovery:
...
 
-Delete the sensor by sending an empty message.
+Delete the sensor by sending an empty message:
 
  ```bash
 mosquitto_pub -h 127.0.0.1 -p 1883 -t "homeassistant/binary_sensor/garden/config" -m ''
 ```
 
-For more details please refer to the [MQTT testing section](/integrations/mqtt/#testing-your-setup).
+Note: the Configuration topic prefix must be `homeassistant` (or another if you changed the default). However, the State topic prefix does not need to be `homeassistant`. It is common for the State topic prefix to indicate which device or script sent the MQTT message, for example 'my_esp_device', 'rtl_433', or 'my_python_script'.
```
Comment (jbouwh) [inline source/_integrations/mqtt.markdown:590]: This note is not needed as it is well documented already.

File: source/_integrations/mqtt.markdown [hunk 3]
```diff
@@ -559,44 +559,233 @@ The following software has built-in support for MQTT discovery:
...
+
+#### Creating a device with two sensor entities (temperature and humidity sensor)
+
+In the following example, a device will be created with two sensor entities representing temperature and humidity measurements. Sensor entities can be represented by an [MQTT Sensor](/integrations/sensor.mqtt/).
+
+To set up a any device with multiple entities, a Configuration topic must be published for each entity. Each Configuration topic must include a "unique_id" property which is unique to the entity being created, and a "device" property which includes a device "identifiers" property which is unique to the device.
+
+A user, script, or device would send its configuration as JSON payload to the Configuration topic. After the first message to `config`, then the device may send MQTT messages with new sensor measurements in the form of a JSON payload.
+
+Set up a device with two sensor entities:
```
Comment (jbouwh) [inline source/_integrations/mqtt.markdown:600]: It is not af any value here to add an example for adding two entities

File: source/_integrations/mqtt.markdown [hunk 4]
```diff
@@ -559,44 +559,233 @@ The following software has built-in support for MQTT discovery:
...
+
+- Configuration topic 1: `homeassistant/sensor/my_custom_sensor_001_temperature/config`
+- Configuration payload 1:
+```json
+{
+    "name": "My Custom Sensor 001 Temperature",
+    "unique_id": "my_custom_sensor_001_temperature",
+    "state_topic": "my_state_topic/my_custom_sensor/001",
+    "value_template": {% raw %} "{{ value_json.temperature }}" {% endraw %},
```
Comment (jbouwh) [inline source/_integrations/mqtt.markdown:609]: Full examples are on the mqtt platform pages, not here please

File: source/_integrations/mqtt.markdown (hunk 2 above)
Comment (raythefourth) [reply source/_integrations/mqtt.markdown:590]: Perhaps the next example is sufficient to illustrate that the 'homeassistant' prefix is not required to be part of the State topic. Would you agree?

File: source/_integrations/mqtt.markdown (hunk 1 above)
Comment (raythefourth) [reply source/_integrations/mqtt.markdown:564]: Would you prefer the title to say "Creating a sensor entity (motion detection binary sensor)" and for the following text to say "In the following example, an entity will be created representing the state of a motion detection sensor. The motion detection sensor can be represented by an [MQTT Binary Sensor](/integrations/binary_sensor.mqtt/)."?

File: source/_integrations/mqtt.markdown (hunk 3 above)
Comment (raythefourth) [reply source/_integrations/mqtt.markdown:600]: I believe there is a lot of value in showing the user how to create a device within Home Assistant that has 2 or more entities. Physical devices usually have more than 1 measurement (for example temperature, humidity, and battery level) and it make sense to group them as a device in Home Assistant. Grouping the entities into a device makes it very easy to see the measurements together on the device page. Wouldn't you agree?

File: source/_integrations/mqtt.markdown (hunk 4 above)
Comment (raythefourth) [reply source/_integrations/mqtt.markdown:609]: This example already existed on the MQTT Configuration page [here](https://www.home-assistant.io/integrations/mqtt/#sensors). The existing example does not work however. I have improved the the example and cleaned up the formatting. 
I believe it is important for Discovery Examples to be found here on this page. These examples will greatly benefit beginners, and this is the first place users will look for information related to Discovery Messages. What do you think?

File: source/_integrations/mqtt.markdown (hunk 2 above)
Comment (jbouwh) [reply source/_integrations/mqtt.markdown:590]: I don't like the big changes in this PR. Can you split typo's small corrections. Larger changes should not be all together as this is hard to review. A lot of the stuff you add I don't want at this page or I do not think changes are needed. I'll do a closer review later.

File: source/_integrations/mqtt.markdown (hunk 3 above)
Comment (jbouwh) [reply source/_integrations/mqtt.markdown:600]: May be, but the examples need to be very simple

File: source/_integrations/mqtt.markdown (hunk 4 above)
Comment (jbouwh) [reply source/_integrations/mqtt.markdown:609]: I'll have a  look. You can also open an issue on that

File: source/_integrations/mqtt.markdown (hunk 1 above)
Comment (jbouwh) [reply source/_integrations/mqtt.markdown:564]: I would prefer keeping this as it is.

File: source/_integrations/mqtt.markdown [hunk 5]
```diff
@@ -559,44 +559,233 @@ The following software has built-in support for MQTT discovery:
...
+#### Creating a single sensor entity (motion detection binary sensor)
 
-A motion detection device which can be represented by a [binary sensor](/integrations/binary_sensor.mqtt/) for your garden would send its configuration as JSON payload to the Configuration topic. After the first message to `config`, then the MQTT messages sent to the state topic will update the state in Home Assistant.
+In the following example, a single entity will be created representing the state of a motion detection sensor. The motion detection sensor has only 2 states, therefore it can be represented by an [MQTT Binary Sensor](/integrations/binary_sensor.mqtt/).
+
+A user, script, or device would send its configuration as JSON payload to the Configuration topic. After the first message to `config`, then the MQTT messages sent to the state topic will update the motion detection sensor state in Home Assistant.
 
 - Configuration topic: `homeassistant/binary_sensor/garden/config`
-- State topic: `homeassistant/binary_sensor/garden/state`
-- Payload:  `{"name": "garden", "device_class": "motion", "state_topic": "homeassistant/binary_sensor/garden/state"}`
```
Comment (jbouwh) [inline source/_integrations/mqtt.markdown:568]: Why removing these lines?

File: source/_integrations/mqtt.markdown [hunk 6]
```diff
@@ -559,44 +559,233 @@ The following software has built-in support for MQTT discovery:
...
+- Configuration payload:  `{"name": "garden", "device_class": "motion", "state_topic": "homeassistant/binary_sensor/garden/state"}`
```
Comment (jbouwh) [inline source/_integrations/mqtt.markdown:569]: The payload is already in the example.

File: source/_integrations/mqtt.markdown [hunk 7]
```diff
@@ -559,44 +559,233 @@ The following software has built-in support for MQTT discovery:
...
 ```
 
-Delete the sensor by sending an empty message.
+Delete the sensor by sending an empty message:
 
  ```bash
 mosquitto_pub -h 127.0.0.1 -p 1883 -t "homeassistant/binary_sensor/garden/config" -m ''
 ```
 
-For more details please refer to the [MQTT testing section](/integrations/mqtt/#testing-your-setup).
```
Comment (jbouwh) [inline source/_integrations/mqtt.markdown:589]: Why is this line removed?

File: source/_integrations/mqtt.markdown (hunk 2 above)
Comment (jbouwh) [inline source/_integrations/mqtt.markdown:590]: This is explained in the settings section.

File: source/_integrations/mqtt.markdown (hunk 4 above)
Comment (jbouwh) [inline source/_integrations/mqtt.markdown:609]: Lets just add add unique_id and a shared device config to the current example, no need to replace the current ones.

File: source/_integrations/mqtt.markdown [hunk 8]
```diff
@@ -559,44 +559,233 @@ The following software has built-in support for MQTT discovery:
...
+Delete this example device and entities:
+- Configuration topic 1: `homeassistant/sensor/my_custom_sensor_001_temperature/config`
+- Configuration payload 1: (empty)
+- Configuration topic 2: `homeassistant/sensor/my_custom_sensor_001_humidity/config`
+- Configuration payload 2: (empty)
+- The messages must be published 'retain' set to true. Do not tick 'Allow template'.
+- If using mosquitto_pub, see previous example.
+
+#### Using a script to create a device with multiple sensor entities (door/window sensors)
+
```
Comment (jbouwh) [inline source/_integrations/mqtt.markdown:651]: Let us not add this here

**Review by jbouwh — CHANGES_REQUESTED** [review]
Sorry to say, but I am not happy with this PR in it current state. We want to keep the documentation compact. May be just add a unique id and device config to the current example.

File: source/_integrations/mqtt.markdown (hunk 8 above)
Comment (jbouwh) [reply source/_integrations/mqtt.markdown:651]: This is some to publish on the Home Assistant community, but I assume you already did that.

**Comment by jbouwh** [comment]
I opened https://github.com/home-assistant/home-assistant.io/pull/28026 to do an alternative suggestion for the parts I do support from your PR.

**Comment by jbouwh** [comment]
I have adjusted some of the topics you addressed with #28026. I do nit think we should include a device creation script in the MQTT docs as we have yaml for that.
@reggleston4  I am going to close this PR, thanx for your effort in this!
